packets1/packets1_test.go

```diff
@@ -21,3 +23,14 @@ func testPacketMarshal(t *testing.T, pkt1 pkts.Packet) pkts.Packet {
 
 	return pkt2
 }
+
+func TestUnmarshalInvalidPacketType(t *testing.T) {
+	buff := bytes.NewBuffer([]byte{
+		2,          // Length
+		byte(0x19), // invalid MsgType
+	})
+	_, err := ReadPacket(buff)
+	if assert.Error(t, err) {
+		assert.Equal(t, err.Error(), "invalid MQTT-SN packet type")
```

Comment:
I wouldn’t check the message as it ties the test to the implementation quite closely and the exact message is presumably not part of the function contract.

Also, in [two other instances](https://github.com/energomonitor/bisquitt/blob/7b7a634fb0f41d2bc68c73efe58412b2bc5236a8/topics/predefined_topics_test.go#L152..L158) where `assert.Error` is used in Bisquitt code, the message is also not checked.

Reply:
This is a pain with Go errors. There's no good mean how to check the error is "the right error". If you only check there's _some_ error, you can easily get false negative result of the test.

In the instances you link, it's probably not a problem but in "complex" operations where several errors can occur, it could be.

I think, it's better to check the error _somehow_ than to not check it at all. In some places, I do not use equality check but rather a `Contains` check:

```
if assert.Error(t, err) {
	assert.Contains(t, err.Error(), "bad AUTH packet length")
}
```
(still uncommitted code of an `Auth` unit test)

Reply:
Understood. Personally, I’d take somewhat imprecise test over coupling it with the implementation so tightly, but if you prefer the other way, I’m fine with that. It most likely won’t lead to any real problems.

Reply:
I prefer the coupling because we have already faced problems with false negative tests in the past.